src/cache/cache_data.h

```diff
@@ -6,15 +6,15 @@
 
 #ifndef CACHE_DATA_H
 #define CACHE_DATA_H
-#include "./result_set_cache_includes.h"
+#include "./cache_includes.h"
 
 /**
  * @brief  struct for holding cache data
  */
 typedef struct CacheData
```

Comment:
I think cache_data.h and cache.h are tightly coupled, please merge into cache.h to reduce overall number of source files

Reply:
moved to cache_includes.h. will not move to cache.h to avoid cyclically inclusions